src/integrationtests/java/com/aws/greengrass/integrationtests/deployment/DeploymentConfigMergingTest.java

```diff
@@ -709,6 +736,85 @@ void GIVEN_kernel_running_service_WHEN_run_with_change_THEN_service_restarts() t
+        kernel.getContext().put(DeviceConfiguration.class, deviceConfiguration);
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel, getClass().getResource("config.yaml"));
+
+        // WHEN
+        CountDownLatch mainRestarted = new CountDownLatch(1);
+        kernel.getContext().addGlobalStateChangeListener((service, oldState, newState) -> {
+            if (service.getName().equals("main") && newState.equals(State.RUNNING) && oldState.equals(State.STARTING)) {
+                mainRestarted.countDown();
+            }
+        });
+
+        // Mock Fss trigger at kernel launch
+        when(mqttClient.publish(any(PublishRequest.class))).thenReturn(CompletableFuture.completedFuture(0));
+
+        // Inject and launch
+        kernel.getContext().put(MqttClient.class, mqttClient);
+        kernel.launch();
+
+        Topics t = kernel.findServiceTopic(FleetStatusService.FLEET_STATUS_SERVICE_TOPICS);
+        assertNotNull(t, "FSS Topics should not be null before merging");
+
+        // Config to merge
+        Map<String, Object> newConfig = ConfigPlatformResolver
+                .resolvePlatformMap(getClass().getResource("delta.yaml"));
+        Map<String, Object> nucleusConfig = getNucleusConfig();
+        ((Map<String, Object>)newConfig.get(SERVICES_NAMESPACE_TOPIC)).put(DEFAULT_NUCLEUS_COMPONENT_NAME,
+                nucleusConfig);
+        Map<String, Object> config =  (Map<String, Object>) nucleusConfig.get(CONFIGURATION_CONFIG_KEY);
+        // Mqtt client config change
+        int expectedMqttPort = 8080;
+        config.put(DeviceConfiguration.DEVICE_MQTT_NAMESPACE, new HashMap<String, Object>() {{
+            put(MqttClient.MQTT_PORT_KEY, expectedMqttPort);
+        }});
+        // Http client config change
+        int expectedDataPlanePort = 443;
+        config.put(DeviceConfiguration.DEVICE_PARAM_GG_DATA_PLANE_PORT, expectedDataPlanePort);
+
+        // Mock Mqtt Validation
+        when(mqttClient.createMqttConnectionBuilder(any(), eq(null))).thenReturn(awsIotMqttConnectionBuilder);
+        when(awsIotMqttConnectionBuilder.withClientId(any())).thenReturn(awsIotMqttConnectionBuilder);
+        when(awsIotMqttConnectionBuilder.build()).thenReturn(mqttClientConnection);
+        CompletableFuture<Boolean> mqttConnection = new CompletableFuture<>();
+        mqttConnection.complete(true);
+        when(mqttClientConnection.connect()).thenReturn(mqttConnection);
+
+        // Mock Http Validation
+        ImmutablePair<SdkHttpClient, GreengrassV2DataClient> pair = new ImmutablePair<>(null, greengrassV2DataClient);
+        when(gscFactory.createClientFromConfig(any())).thenReturn(pair);
+        when(greengrassV2DataClient.listThingGroupsForCoreDevice((ListThingGroupsForCoreDeviceRequest) any()))
+                .thenReturn(null);
+
+        // Set mocks
+        deploymentConfigMerger.setMqttClient(mqttClient);
+        deploymentConfigMerger.setFactory(gscFactory);
```

Comment:
There was an issue with deploymentConfigMerger not picking up the mocks. After a quick look I believe it's because the class is initialized and has the fields injected before the mocks are placed into the context.

I will refactor this to have the mocks injected in the before() function call